azure-go-aks/main.go

```diff
@@ -13,6 +13,27 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 )
 
+func getKubeconfig(ctx *pulumi.Context, cluster *containerservice.ManagedCluster, resourceGroup *resources.ResourceGroup) pulumi.Output {
```

Comment:
Broke this out into a separate function to break apart the instantiation logic and the kubeconfig logic.